sci-libs/rocRAND/rocRAND-5.7.1.ebuild

```diff
@@ -0,0 +1,71 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+ROCM_VERSION=${PV}
+
+inherit cmake flag-o-matic rocm
+
+DESCRIPTION="Generate pseudo-random and quasi-random numbers"
+HOMEPAGE="https://github.com/ROCmSoftwarePlatform/rocRAND"
+SRC_URI="https://github.com/ROCmSoftwarePlatform/${PN}/archive/rocm-${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+KEYWORDS="~amd64"
+SLOT="0/$(ver_cut 1-2)"
+IUSE="benchmark test"
+REQUIRED_USE="${ROCM_REQUIRED_USE}"
+
+# test are failing (90% tests passed, 4 tests failed out of 40)
+RESTRICT="test"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-5.4.2_stdint-gcc13.patch
+	"${FILESDIR}"/${PN}-5.7.1_fix_generator_alignment.patch
+)
+
+RDEPEND="dev-util/hip"
+DEPEND="${RDEPEND}
+dev-util/rocm-cmake
+test? ( dev-cpp/gtest )"
+BDEPEND="dev-util/rocm-cmake
+>=dev-util/cmake-3.22"
+
+S="${WORKDIR}/rocRAND-rocm-${PV}"
+
+src_configure() {
+	addpredict /dev/kfd
+	addpredict /dev/dri/
+
+	# hipcc compiles everything with -O3; other modes cause clang backend errors
+	filter-flags -O?
```

Comment:
I cannot reproduce `clang backend` errors. Maybe you should file a bug to llvm upstream. I would suggest put this as a ewarn, to remind people turning off `-O` flag manually if they encounter clang backend errors.

Reply:
> I cannot reproduce `clang backend` errors.

Retried after system update (basically, clang 17.0.2 -> 17.0.3 and now it worked without removing -O2).
There is still an open question: is it a valid assumption, that it is good for end-user to use global CFLAGS for compiling GPU applications? It is a well-known fact, for example, that exactly -O3 mode provides the best result for CUDA in terms of registry pressure. Should not there be `GPU_CFLAGS_*` in pair to https://wiki.gentoo.org/wiki/CPU_FLAGS_* ?
Anyways, this is just an open question. For now the target is just to stabilize.

Reply:
>  Anyways, this is just an open question. For now the target is just to stabilize.

You are right, currently there's no detailed documentation on how to choose GPU optimization flag. Also, I did not find a way to set clang optimization level separately for CPU and GPU code generation. So currently let's strip the `-O` flag and use `-O3` which is the default set by hipcc

Reply:
> it outputs nothing

Currently clang does not properly detects the rocm installation located at `/usr` in Gentoo. So `hipcc` manually parsed `--rocm-path=${EPREFIX}/usr` in the perl script.